protocol/shannon/fullnode_cache.go

```diff
@@ -34,138 +33,136 @@ const (
 	// TODO_MAINNET_MIGRATION: Ensure applications are invalidated during unstaking. Revisit these values after mainnet migration to ensure no race conditions.
 	defaultAppCacheTTL = 5 * time.Minute
 
-	// Cleanup interval is twice the TTL.
-	// standard practice to balancing cleanup with minimal processing overhead
-	defaultAppCacheCleanupInterval = defaultAppCacheTTL * 2
-
 	// As of #275, on Beta TestNet.
 	// - Blocks are 30 seconds
 	// - A session is 50 blocks
 	// - A grace period is 1 block (i.e. 30 seconds)
 	// TODO_MAINNET_MIGRATION: Revisit these values after mainnet migration to ensure no race conditions.
 	defaultSessionCacheTTL = 30 * time.Second
-	
-	// Cleanup interval is twice the TTL.
-	// Standard practice for balancing cleanup with minimal processing overhead
-	defaultSessionCacheCleanupInterval = defaultSessionCacheTTL * 2
 
-	// Cache key prefixes to avoid collisions
+	// Early refresh configuration for apps
+	// Apps will start refreshing when they're between 4-4.5 minutes old
+	appMinRefreshDelay = 4 * time.Minute
+	appMaxRefreshDelay = 4*time.Minute + 30*time.Second
+
+	// Early refresh configuration for sessions
+	// Sessions will start refreshing when they're between 20-25 seconds old
+	sessionMinRefreshDelay = 20 * time.Second
+	sessionMaxRefreshDelay = 25 * time.Second
+
+	// Retry base delay for exponential backoff on failed refreshes
+	retryBaseDelay = 100 * time.Millisecond
+
+	// Cache configuration
+	cacheCapacity      = 100_000
+	numShards          = 10
+	evictionPercentage = 10
+)
+
+const (
 	appCacheKeyPrefix     = "app"
 	sessionCacheKeyPrefix = "session"
 )
 
 var _ FullNode = &cachingFullNode{}
 
 // cachingFullNode implements the FullNode interface by wrapping a LazyFullNode
-// and caching results to improve performance.
+// and caching results to improve performance with automatic refresh-ahead.
+//
+// Early Refresh Strategy:
+// Uses SturdyC's early refresh to prevent thundering herd and eliminate latency spikes.
+// Background refreshes happen before entries expire, so GetApp/GetSession never block.
+//
+//   - Apps: 5min TTL, refresh at 4-4.5min (80-90% of TTL)
```

Comment:
Clarify that this is an example. Numbers will change. Add TODO on me.

Reply:
Updated.